### PR TITLE
Task visibility

### PR DESCRIPTION
**Pull request by willdoran**
This pull request makes the following changes:
-

Testing checklist:
- [ ]

Fixes ushahidi/platform#1945

Ping @ushahidi/platform

### CODE REVIEW

File: app/main/posts/modify/post-tabs.html [hunk 1]
```diff
@@ -2,6 +2,14 @@
     <header class="form-sheet-summary">
         <h1 class="form-sheet-title section-title">{{stage.label}}</h1>
     </header>
+    <div ng-show="stage.task_is_internal_only" class="alert">
+        <p>This task is marked for internal use only. It is not seen by users when they create a post.</p>
```
Comment (rjmackay) [inline app/main/posts/modify/post-tabs.html]: Need to translate

File: app/main/posts/modify/post-tabs.html [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+    </div>
+    <div>
+        <div ng-show="stage.show_when_published" class="alert">
```
Comment (rjmackay) [inline app/main/posts/modify/post-tabs.html]: Added in #731

**Comment by rjmackay** [comment]
@willdoran @jshorland I suspect this should go in for 2chaguzi right?

File: app/settings/surveys/survey-editor.directive.js [hunk 3]
```diff
@@ -122,7 +122,8 @@ function SurveyEditorController(
                         priority: 0,
                         required: false,
                         type: 'post',
-                        show_when_published: 1,
+                        show_when_published: false,
```
Comment (rowasc) [inline app/settings/surveys/survey-editor.directive.js]: Note: this default is different than what we have in develop . Develop has `true` instead for show_when_published.

**Comment by rowasc** [comment]
What seems to be missing in develop is locks for post-edit-controller.js the rest is already integrated.
post-edit-controller is not used anymore since we do everything with the post.data.edit controller. CLOSING

**Pull request closed without merging**